internal/statemachine/helper_test.go

```diff
@@ -335,6 +335,8 @@ func TestCleanup(t *testing.T) {
 			t.Errorf("Error: temporary workdir %s was not cleaned up\n",
 				stateMachine.stateMachineFlags.WorkDir)
 		}
+		// remove a leftover artifact
+		os.Remove("ubuntu-image.gob")
```

Comment:
Can you help me understand why this one is needed?  As I understand it, this file is created in WorkDir, and we've just established that WorkDir doesn't exist?

Would a defer before the assert be better to ensure that the cleanup happens, even on test failure?

Reply:
Keen eye, you've made me realize TestCleanup isn't actually testing the cleanup properly. I'll push an update that fixes that and simultaneously removes the need for an explicit remove.